### PR TITLE
CT 1509 reorg protobuf events to have top-level Event message

### LINKED ISSUE #6256: [CT-1509] Reorganize protobuf structured logging events
As per discussion with Dave Berton, attempt to reorganize the existing protobuf messages so that instead of an EventInfo submessage, those fields will be moved to a top-level Event, and the rest of the logging events will be submessages specified in a "oneof" in the Event message. This should make it easier to deserialize. 

The current creation of events for the "fire_event" function should be able to stay the same, except for tweaks (improvements) to the way that we set "level". Instead of setting the event keys in the a post_init of the BaseEvent class, we will construct an Event in "fire_event" and set the passed in message as the "detail" part of the event.

This is part of the structured logging initiative, such as we see in https://github.com/dbt-labs/dbt-core/pull/6136.

### PR DESCRIPTION
**Pull request by gshank**
resolves #6256

### Description

Reorganize the protobuf logging events so that the top level is the common Event message, which will contain a "oneof" of the entire list of detailed events.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Comment by gshank** [comment]
Note: there is a lot of noise in this pull request because of the need to touch every one of our logging events, fix the tests, etc. The significant parts regarding protobuf message organization are in the core/dbt/events/types.proto and core/dbt/events/event.proto files.

**Comment by gshank** [comment]
Looks like there might be one of those logging-error-causing-hang problems now. Not happening with a local run, of course.

File: core/dbt/events/functions.py [hunk 1]
```diff
@@ -116,23 +115,35 @@ def stop_capture_stdout_logs() -> None:
...
+        # need defaults in the json log output, but that includes a huge unnecessary
+        # pile of oneof field dictionaries, so we'll have to munge it.
         event_dict = event.to_dict(casing=betterproto.Casing.SNAKE, include_default_values=True)  # type: ignore
     except AttributeError as exc:
         event_type = type(event).__name__
-        raise Exception(f"type {event_type} is not serializable. {str(exc)}")
-    return event_dict
+        raise BaseException(f"type {event_type} is not serializable. {str(exc)}")
+
+    # Kludge warning: create a new dictionary with only the relevant keys. In the
```
Comment (emmyoop) [inline core/dbt/events/functions.py:141]: Can you create an issue to investigate this and link to the issue in the comment?  That way it's less likely to get lost and when we read this we can know where to find more context.

Comment (gshank) [reply core/dbt/events/functions.py:141]: Created a ticket and added a comment.

**Comment by gshank** [comment]
The performance of this approach has proved to be unacceptable, so I'm closing this PR.

**Pull request closed without merging**